src/components/Dashboard/DashboardCard/DashboardCardDropdownItem.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'patternfly-react';
+
+import { getFormElement } from '../../Form';
+import { DROPDOWN } from '../../Form/constants';
+
+export class DashboardCardDropdownItem extends React.PureComponent {
+  render() {
+    const { id, dropdownOptions, onDropdownChange, dropdownType, selectedOption } = this.props;
+
+    const filterDropdown = {
+      id,
+      type: DROPDOWN,
+      choices: dropdownOptions,
+      onChange: newValue => onDropdownChange(dropdownType, newValue),
+      value: selectedOption,
+      className: 'kubevirt-dashboard-card-dropdown',
+    };
+    return getFormElement(filterDropdown);
```

Comment:
can you please create the dropdown element directly here instead of calling `getFormElement`. It is easier to read.

Reply:
Not sure if this component is needed at all. Can't we just use this inside TopConsumers? I see that the only difference is the classname. Also not sure if DashboardCardDropdownBody is necessary.

Reply:
@suomiy There is a function 'getFormElement` inside FormFactory.js file which takes care of creating all the form components. So, I don't feel there is a need to rewrite code for creating dropdown component.

`DashboardCardDropdownItem` is a generic component which needs to be used across various cards in OCS dashboard for eg. Top Consumer, Performance, Capacity, Alerts. Hence, I have created a component. There as some cards where we need more than one dropdown(i.e. 2 or 3). `DashboardCardDropdownBody ` is parent component where you can pass as many dropdowns you required for the card(like we have done for Utilization card).

Reply:
> There is a function 'getFormElement` inside FormFactory.js file which takes care of creating all the form components. So, I don't feel there is a need to rewrite code for creating dropdown component.

I think it would help the code readability because It just obfuscate things. It is not about rewriting the code for creating dropdown component. It is about using the dropdown component.

> DashboardCardDropdownItem is a generic component which needs to be used across various cards in OCS dashboard for eg. Top Consumer, Performance, Capacity, Alerts. Hence, I have created a component. There as some cards where we need more than one dropdown(i.e. 2 or 3). DashboardCardDropdownBody is parent component where you can pass as many dropdowns you required for the card(like we have done for Utilization card).

I don't think `width: 36% !important;` is specifically reusable. And I couldn't find the css for `kubevirt-dashboard-card-dropdown`. 

If it is made reusable as especially styled component, then please let these components in.

Reply:
@suomiy ok. Do you mean to write the following code -
` 
<Dropdown
          id={id}
          value={value || defaultValue}
          onChange={onChange}
          choices={choices}
          disabled={disabled}
          className={className}
        />
`
with values populated in the component?
I feel, the intention of the `getFormElement` function is to return the form elements based on the type passed. @rawagner Please correct me, if I am wrong.  Just exploring the better approach to do it.

 Also,if you refer https://github.com/kubevirt/web-ui-components/blob/master/src/components/ClusterOverview/TopConsumers/TopConsumers.js#L123, the same way dropdown component has been created.

Reply:
@gnehapk @suomiy I dont mind using `getFormElement`. We should remove export from `getFormElement` if we dont want to use it outside `FormFactory`.

Reply:
> Do you mean to write the following code -
`<Dropdown id={id} value={value || defaultValue} onChange={onChange} choices={choices} disabled={disabled} className={className} />`

yes, with the props you need

> We should remove export from getFormElement if we dont want to use it outside FormFactory

> I feel, the intention of the getFormElement function is to return the form elements based on the type passed. @rawagner Please correct me, if I am wrong. Just exploring the better approach to do it.

this is mainly useful in `EditableDraggableTable` until that is refactored to pf4 - then we will see. So we need the export. 

I think using the underlying elements is preferable when dynamic approach is not required - i.e. this use case

> Also,if you refer https://github.com/kubevirt/web-ui-components/blob/master/src/components/ClusterOverview/TopConsumers/TopConsumers.js#L123, the same way dropdown component has been created.

I guess this would be also good to unify. Could you please include this in this PR?  I suspect TopConsumers should be similar for both dashboards, right?